### ISSUE TITLE
Does installer caching work on GitHub-hosted runners?

### ISSUE THREAD

**Issue by AndreasAlbertQC** [conda-incubator/setup-miniconda#238]
In the context of #230, I attempted to use installer caching with GitHub-hosted runners (i.e. not self-hosted). It appears to me that in this case, installer caching never works. Does it work for anyone else?

Example workflow is [here](https://github.com/AndreasAlbertQC/setup-miniconda-cache-test/runs/6510519479). There are two jobs in the same workflow doing nominally the same thing. Under the "Ensuring installer" heading in the logs, the first job logs this statement:

```
Can we download a custom installer by URL?
  ... will download a custom installer by URL.
  Ensuring Installer...
  Checking for cached Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh@0.0.0-67[18](https://github.com/AndreasAlbertQC/setup-miniconda-cache-test/runs/6510519479?check_suite_focus=true#step:3:21)bb84a08e4157b82f63e60d990e450fc66f560a4591c0d3fb1c0[19](https://github.com/AndreasAlbertQC/setup-miniconda-cache-test/runs/6510519479?check_suite_focus=true#step:3:22)a4f2ee1...
  Did not find Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh 0.0.0-6718bb84a08e4157b82f63e60d990e450fc66f560a4591c0d3fb1c019a4f2ee1 in cache
  Downloaded Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh, ensuring extension .sh
  Caching Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh@0.0.0-6718bb84a08e4157b82f63e60d990e450fc66f560a4591c0d3fb1c019a4f2ee1...
  Cached Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh@0.0.0-6718bb84a08e4157b82f63e60d990e450fc66f560a4591c0d3fb1c019a4f2ee1: /opt/hostedtoolcache/Miniforge-pypy3-4.8.3-2-Linux-x86_64.sh/0.0.0-6718bb84a08e4157b82f63e60d990e450fc66f560a4591c0d3fb1c019a4f2ee1/x64!
```

So the first job does not see a cache, downloads the installer, and then caches it. This is expected behavior. However, the second job then proceeds to log exactly the same messages. It also does not see the cache and proceeds to re-download. I had instead expected the second job to pick up the cached file from the first job.

Bottom line: Is the cache location (`/opt/hostedtoolcache/`) never persistent between jobs? Only sometimes, and I got unlucky?

**Comment by goanpeca** [comment]
Do you think this is something that needs to be raised with github actions or something we need to address here?

**Comment by AndreasAlbertQC** [comment]
Maybe there is no need to even implement the caching here at all? One could outsource caching to the [cache action](https://github.com/actions/cache), which AFAICT is the accepted way of caching such data. I have not used it myself yet, so I am not familiar enough with the usage to give an example here, but it should be feasible without too much overhead.
